eng/pipelines/swagger-api-doc-preview.yml

```diff
@@ -0,0 +1,203 @@
+          # Path does not need to be set because sparse-checkout.yml already
+          # checks out files in the repo root
+          Repositories:
+            - Name: MicrosoftDocs/AzureRestPreview
+              # TODO: use main branch when this is ready to go live
+              Commitish: refs/heads/doc-preview-migration-test
+              WorkingDirectory: AzureRestPreview
+
+      - template: /eng/pipelines/templates/steps/npm-install.yml
+
+      - script: npm exec --no -- api-doc-preview --output ../../AzureRestPreview
+        displayName: Generate Swagger API documentation preview
+        workingDirectory: .github/shared
+
+      - template: /eng/common/pipelines/templates/steps/git-push-changes.yml
+        parameters:
+          WorkingDirectory: AzureRestPreview
+          BaseRepoOwner: MicrosoftDocs
+          TargetRepoOwner: MicrosoftDocs
+          TargetRepoName: AzureRestPreview
+          BaseRepoBranch: $(BranchName)
+          CommitMsg: |
+            Update API Doc Preview
+            Build: $(System.CollectionUri)$(System.TeamProject)/_build/results?buildId=$(Build.BuildId)
+            PR: $(System.PullRequest.SourceRepositoryURI)/pull/$(System.PullRequest.PullRequestId)
+
+      - task: AzureCLI@2
+        displayName: Start docs build
+        inputs:
+          azureSubscription: msdocs-apidrop-connection
+          scriptType: pscore
+          scriptLocation: inlineScript
+          inlineScript: |
+            $buildStartRaw = az pipelines build queue `
+              --organization "https://dev.azure.com/apidrop/" `
+              --project "Content CI" `
+              --definition-id "8120" `
+              --variables 'params={"target_repo":{"url":"https://github.com/MicrosoftDocs/AzureRestPreview","branch":"$(BranchName)"}, "source_of_truth": "code"}'
+            $buildStartRaw | Set-Content buildstart.json
+            $buildStart = $buildStartRaw | ConvertFrom-Json
+            Write-Host "Build started at https://dev.azure.com/apidrop/Content%20CI/_build/results?buildId=$($buildStart.id)"
+
+      - template: /eng/pipelines/templates/steps/set-pr-check.yml
+        parameters:
+          State: pending
+          TargetUrl: $(CurrentBuildUrl)
+          Description: 'Waiting for docs build to finish'
+          Context: $(StatusName)
+
+      - task: AzureCLI@2
+        displayName: Wait for docs build to finish
+        # Retry on failure to handle transient issues or builds that run longer
+        # than the token used by az CLI is valid
+        retryCountOnTaskFailure: 3
+        inputs:
+          azureSubscription: msdocs-apidrop-connection
+          scriptType: pscore
+          scriptLocation: inlineScript
+          inlineScript: |
+            $buildStart = Get-Content buildstart.json | ConvertFrom-Json
+
+            Write-Host "Waiting for build to finish: https://dev.azure.com/apidrop/Content%20CI/_build/results?buildId=$($buildStart.id)"
+            
+            while($true) { 
```

Comment:
Usually better to limit to a number of retries or a time limit (X minutes).  Otherwise could hang until we hit pipeline timeout.